### ISSUE TITLE
Voicecall recording issues

### ISSUE THREAD

**Issue by PaskoMoto** [mer-hybris/pulseaudio-modules-droid#67]
Hi @jusa 
I'm experiencing issues while recording calls natively in my Xperia X with SailFish X.
The problem is that the audio from the voice calls is not recorded properly, while the audio recorded while I speak to the mic (without call established) is recorded fine.
I've been tracing with pulseaudio and seems to me that there could be a problem with <module-droid-card.c>, the driver for <source.primary> in my device configuration.

Here's how I tested:
```

[nemo@Sailfish CallRecordings]$ pactl set-card-profile 0 voicecall-record
[nemo@Sailfish CallRecordings]$ parec --record --device=sink.primary.monitor --file-format=wav --verbose output.wav
Opening a recording stream with sample specification 's16le 2ch 44100Hz' and channel map 'front-left,front-right'.
Connection established.
Stream successfully created.
Buffer metrics: maxlength=4194304, fragsize=352800
Using sample spec 's16le 2ch 44100Hz', channel map 'front-left,front-right'.
Connected to device source.primary (index: 2, suspended: no).
Stream moved to device source.primary (2, not suspended).
Got signal, exiting.ncy: 18843 usec.
```

My test consists in recording, first a voicecall, then hangup the call and speak directly to the microphone (this is reflected by the "Stream moved to device..." message in the log I attached).
When I listen to the recorded file the part corresponding to the voicecall is wrong and the part where I speak to the microphone is OK.
I can provide a sample if you think it would be useful.
Also, there's a question opened in TJC and this may be a redundant request... let me know :-D
https://together.jolla.com/question/170695/call-recording-broken-audio/
Regards.

**Comment by jusa** [comment]
Hi, and thanks for the report.

That TJC link you pasted is about the same issue.

The issue is actually that with Xperia we open the input stream with wrong channel count, which then causes duplicate samples and thus 50% slower playback rate. I'll prepare a pull request shortly.

**Comment by jusa** [comment]
@PaskoMoto rpms with the fix here:

https://kraa.fi/tmp/pulseaudio-modules-droid-8.0.62-1.armv7hl.rpm
https://kraa.fi/tmp/pulseaudio-modules-droid-common-8.0.62-1.armv7hl.rpm

**Comment by PaskoMoto** [comment]
Hi @jusa 

I have tested it right now and works perfect! Thanks a lot. :-)

Would you mind if I post the download links you provided as an answer in TJC?

Best regards.

**Comment by jusa** [comment]
@PaskoMoto maybe better not to paste the links there, to avoid any issues when users are fiddling with their devices. The fix is included in an upcoming 2.1.3 version anyway, which is going to be released soon(ish).

**Comment by PaskoMoto** [comment]
Hi.
OK, I understand.
Again thank you for you support, I had no issues whatsoever.
Regards.
